exec/cpg.c

```diff
@@ -1157,10 +1181,24 @@ static void do_proc_join(
 	notify_info.nodeid = nodeid;
 	notify_info.reason = reason;
 
-	notify_lib_joinlist(&pi->group, NULL,
-			    1, &notify_info,
-			    0, NULL,
-			    MESSAGE_RES_CPG_CONFCHG_CALLBACK);
+	if (group_notify_map == NULL) {
+		notify_lib_joinlist(&pi->group, NULL,
+				    1, &notify_info,
+				    0, NULL,
+				    MESSAGE_RES_CPG_CONFCHG_CALLBACK);
+	} else {
+		struct join_list_confchg_data *jld = qb_map_get(group_notify_map, pi->group.value);
+		if (jld == NULL) {
+			jld = (struct join_list_confchg_data *)calloc(1, sizeof(struct join_list_confchg_data));
```

Comment:
memory allocated but not freed.

Reply:
fixed in v2 (free happens when iterating over the map in joinlist_inform_clients)